bottlecap/src/traces/propagation/mod.rs

```diff
@@ -1,3 +1,873 @@
+                let tracestate: Option<String> = match style {
+                    TracePropagationStyle::TraceContext => {
+                        context.tags.get(TRACESTATE_KEY).cloned()
+                    }
+                    _ => None,
+                };
+                let attributes = HashMap::from([
+                    ("reason".to_string(), "terminated_context".to_string()),
+                    ("context_headers".to_string(), style.to_string()),
+                ]);
+                let trace_id_high_str = context
+                    .tags
+                    .get(DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY)
+                    .cloned()
+                    .unwrap_or_default();
+                let trace_ig_high = u64::from_str_radix(&trace_id_high_str, 16).unwrap_or_default();
+
+                links.push(SpanLink {
+                    trace_id: context.trace_id,
+                    trace_id_high: trace_ig_high,
+                    span_id: context.span_id,
+                    flags: u32::from(sampling > 0),
+                    tracestate: tracestate.unwrap_or_default(),
+                    attributes,
+                });
+            } else if style == TracePropagationStyle::TraceContext {
+                if let Some(tracestate) = context.tags.get(TRACESTATE_KEY) {
+                    primary_context
+                        .tags
+                        .insert(TRACESTATE_KEY.to_string(), tracestate.clone());
+                }
+
+                if primary_context.trace_id == context.trace_id
+                    && primary_context.span_id != context.span_id
+                {
+                    let mut dd_context: Option<SpanContext> = None;
+                    if styles.contains(&TracePropagationStyle::Datadog) {
+                        let position = styles
+                            .iter()
+                            .position(|&s| s == TracePropagationStyle::Datadog)
+                            .unwrap_or_default();
+                        dd_context = contexts.get(position).cloned();
+                    }
+
+                    if let Some(parent_id) = context.tags.get(DATADOG_LAST_PARENT_ID_KEY) {
+                        primary_context
+                            .tags
+                            .insert(DATADOG_LAST_PARENT_ID_KEY.to_string(), parent_id.clone());
+                    } else if let Some(sc) = dd_context {
+                        primary_context.tags.insert(
+                            DATADOG_LAST_PARENT_ID_KEY.to_string(),
+                            format!("{:016x}", sc.span_id),
+                        );
+                    }
+
+                    primary_context.span_id = context.span_id;
+                }
+            }
+
+            i += 1;
+        }
+
+        primary_context.links = links;
+
+        primary_context
+    }
+
+    fn attach_baggage(context: &mut SpanContext, carrier: &dyn Extractor) {
```

Comment:
can this method be outside of the struct implementation?

Reply:
Same question as the other comment